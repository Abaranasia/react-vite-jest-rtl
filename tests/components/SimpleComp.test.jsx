const { screen, render } = require("@testing-library/react");
const { SimpleComp } = require("../../src/components/SimpleComp");

describe('SimpleComp tests', () => {

  const testText = "hello world";

  test('SimpleComp snapshot', () => {
    const { container } = render(
      <SimpleComp
        value={testText}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('SimpleComp snapshot', () => {
    render(
      <SimpleComp
        value={testText}
      />
    );

    expect(screen.getByText(testText)).toBeTruthy()
    expect(screen.getByRole('heading', { level: 3 }).innerHTML).toContain(testText);
  });
})