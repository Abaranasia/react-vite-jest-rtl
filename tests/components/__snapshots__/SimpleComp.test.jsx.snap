// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleComp tests SimpleComp snapshot 1`] = `
<div>
  <div
    style="background-color: pink; padding: 5px;"
  >
    <h2>
      Simple Component
    </h2>
    <h3>
       
      hello world
       
    </h3>
  </div>
</div>
`;
